### PR TITLE
Add kiki-mig environment to test db migrations

### PR DESCRIPTION
**Pull request by pivotalgeorge**
the new kiki environment is set up like the old kiki environment
(short-lived, cf-deployment-based, with the use-postgres ops file).

It is intended to ensure db migrations do not break CAPI during upgrade. see the [capi-ci README](https://github.com/cloudfoundry/capi-ci/blob/c31d0bcc78c58a83f85e01428804baeb924ff640/README.md#whats-up-with-kiki)

This changeset includes
- bbl up/down for kiki-mig
- back-compat job that bosh deletes and redeploys cf, runs migrations, and CATS and SITS test suites
- deprecated db ip caching code from apply_latest_migrations removed in `afcc169` but relevant to this work
- add attempts: 5 to SITS run against kiki in back-compat
- gate bump-ci-passed on kiki passing migrations

Key followup work: issue https://github.com/cloudfoundry/capi-ci/issues/49

Co-authored-by: Seth Boyles <seth.boyles@broadcom.com>

### CODE REVIEW

**Review by Benjamintf1 — APPROVED** [review]
lgtm. Just restoring a deleted test. As I understand this was approved by wg.

File: ci/pipeline.yml [hunk 1]
```diff
@@ -957,6 +1102,90 @@ jobs:
...
+      file: cf-deployment-concourse-tasks/bbl-destroy/task.yml
+      ensure:
+        params:
+          rebase: true
+          repository: updated-bbl-state
+        put: kiki-mig-bbl-state
+      input_mapping:
+        bbl-state: kiki-mig-bbl-state
+      params:
+        #BBL_CONFIG_DIR: kiki-mig/bbl-config
```
Comment (moleske) [inline ci/pipeline.yml:1176]: Can we remove these comment out params?

**Review by moleske — APPROVED** [review]
Looks good for matching what we had before the old ci fell over.  One question that's not material to approval

File: ci/pipeline.yml (hunk 1 above)
Comment (Benjamintf1) [reply ci/pipeline.yml]: done

**Pull request merged**